lib/services/systemd.c

```diff
@@ -75,6 +75,32 @@ systemd_new_method(const char *method)
  */
 
 static DBusConnection* systemd_proxy = NULL;
+static svc__systemd_callback_t systemd_callback = NULL;
+static void *systemd_callback_data = NULL;
+
+/*!
+ * \internal
+ * \brief Register a function to be called when a systemd job completes
+ *
+ * If the application registers a callback using this function, the callback
+ * will be called whenever a systemd JobRemoved signal is received. This allows
+ * called whenever a systemd JobRemoved signal is received. This allows
+ * applications to be notified when a systemd action completes, rather than just
+ * when it is initiated.
+ *
+ * \param[in] callback   Function to call when jobs complete
+ * \param[in] user_data  Data to pass to the callback
+ *
+ * \note The callback must be registered before any other systemd function
```

Comment:
only really needs to be registered before we start caring about results of jobs initiated after that point -- which, for the one current user (the executor), is ASAP

Reply:
This comment seems kind of useless.  It's really just trying to say "register a callback before you want to use the callback to process systemd messages" which, of course.

Reply:
Hahaha, agreed.